lib/orocos/process.rb

```diff
@@ -220,26 +230,40 @@ def each_task
 
         # Returns the TaskContext instance for a task that runs in this process,
         # or raises Orocos::NotFound.
-        def task(task_name, name_service = Orocos.name_service)
+        def task(task_name, name_service = nil)
```

Comment:
This new implementation assumes that one is calling `wait_running`, which is not the workflow outside of Syskit

You need to call wait_running if the IOR is not available and try again, or make sure wait_running is a straightforward no-op once we did get all tasks, and call it unconditionally.